Client/Frontend/Fakespot/FakespotViewModel.swift

```diff
@@ -25,15 +25,20 @@ class FakespotViewModel {
         switch viewState {
         case .loading:
             return [.loadingView]
-        case let .loaded(product, analysisStatus, analysisCount):
-            guard let product else {
+        case .loaded(let productState):
+            guard let product = productState.product else {
                 return [
                     .messageCard(.genericError),
                     .qualityDeterminationCard,
                     .settingsCard
                 ]
             }
 
+            let productAdCard = productState
```

Comment:
I remember seeing something about us choosing the ad with the highest rating to display. Let's verify this and change the code accordingly.

Reply:
Here is the desktop ticket for that: https://bugzilla.mozilla.org/show_bug.cgi?id=1863951